test/themes/base.theme.git.bats

```diff
@@ -0,0 +1,390 @@
+#!/usr/bin/env bats
+
+load ../test_helper
+load ../../lib/composure
+
+cite _about _param _example _group _author _version
+
+load ../../lib/helpers
+load ../../themes/githelpers.theme
+load ../../themes/base.theme
+
+add_commit() {
+  local file_name="general-${RANDOM}"
+  touch "${file_name}"
+  echo "" >> "${file_name}"
+  git add "${file_name}"
+  git commit -m"message"
+}
+
+setup_repo_with_upstream() {
+  upstream="$(mktemp -d)"
+  pushd "$upstream" > /dev/null
+  git init . > /dev/null
+  add_commit > /dev/null
+  git checkout -b branch-two
+  git checkout -b gone-branch
+  git checkout master
+  popd > /dev/null
+
+  downstream="$(mktemp -d)"
+  pushd "$downstream" > /dev/null
+  git init . > /dev/null
+  add_commit > /dev/null
+  git remote add my-remote "$upstream"
+  git fetch my-remote
+  git branch -u my-remote/master > /dev/null
+  popd > /dev/null
+
+  pushd "$upstream" > /dev/null
+  git branch -d gone-branch > /dev/null
+  popd > /dev/null
+
+  pushd "$downstream" > /dev/null
+  git fetch my-remote
+  popd > /dev/null
+
+  echo "$downstream"
+}
+
+@test 'themes base: SCM_BRANCH: when tracking a remote branch: it shows the commits ahead and behind' {
+  pre="\$(_git-friendly-ref)"
+
+  remote="$(mktemp -d)"
+  pushd "$remote"
+  git init .
+  add_commit
+  add_commit
+  popd
```

Comment:
You sometimes use `popd > /dev/null` and sometimes just `popd` - any specific reason?

Reply:
The `setup_repo_with_upstream` function returns the directory that the test will use, so it can't echo any additional output. The tests themselves should output, so don't have their output swallowed.